pkg/storage/ingest/pusher.go

```diff
@@ -119,16 +136,23 @@ func (c pusherConsumer) unmarshalRequests(ctx context.Context, records []record,
 	defer close(recC)
```

Comment:
[nit] I would add a comment to `unmarshalRequests()` to explain that `ctx` MUST be canceled once processing of all records is done, otherwise we leak resources. The reason why I think we'll leak resources otherwise, is because `rec.ctx` cancel function is called only once `ctx` is done.